### PR TITLE
Add metric MultiManager and MetricPlotter

### PR DESCRIPTION
**Pull request by rcgorman-dstl**
- Have replaced SimpleManager metric manager with new MultiManager class so this will be a breaking change. 
- MultiManager can take multiple metric generators of the same type and multiple sets of the same type to generate metrics. 
- All metrics that take in track and truth can now also take in 2 tracks.
- Additional functionality added to MultiManager through new methods to help easier extraction of metrics and insights.
- New MetricPlotter class created to make plotting metrics easier.

Update:
- Have added SimpleManager metric manager back in to avert breaking change. It should operate in the same way as before.

### CODE REVIEW

**Review by nperree-dstl — COMMENTED** [review]
Seems to work really well, comments are mostly documentation issues. I think you also need to remove the old `SimpleManager` from test_pcrbmetric and test_uncertaintymetric to stop tests failing.

File: docs/examples/Metrics.py [hunk 1]
```diff
@@ -1,212 +1,485 @@
...
 # %%
-# Once we've created a set of metrics, these are added to a Metric Manager, along with the
-# associator. The associator can be used by multiple metric generators, only being run once as this
-# can be a computationally expensive process; in this case, only SIAP Metrics requires it.
-from stonesoup.metricgenerator.manager import SimpleManager
+# Add data to metric manager and generate metrics
+# -----------------------------------------------
+# Now that we have all of our ground truth, detections, and tracks data, we can add it to the metric manager.
+#
+# The MultiManager .add_data() method takes a dictionary of lists/sets of ground truth, detections, and/or tracks data.
```
Comment (jswright-dstl) [inline docs/examples/Metrics.py:346]:
```suggestion
# The MultiManager :meth:`.add_data()` method takes a dictionary of lists/sets of ground truth, detections, and/or tracks data.
```

**Review by jswright-dstl — APPROVED** [review]

**Review by jswright-dstl — APPROVED** [review]

**Comment by sdhiscocks** [comment]
The breaking change makes this difficult to merge at this point. However, it may be possible to make it backwards compatible fairly easy:
 - Add back in the `SimpleManager`, inheriting from MultiManager, with "tracks", "groundtruth_paths" and "detections" as default keys in `states_set`.
 - Have property on `SimpleManager` for `tracks`, `groundtruth_paths` and `detections` that returns values from those dictionary keys mentioned above.
 - Change metrics so default for `tracks_key` and `truths_key` are "tracks" and "groundtruth_paths".
 - Also provide sensible default `generator_name` for each metric.

I've not tried this, but something like this should work or at least be a good start.

File: stonesoup/metricgenerator/manager.py [hunk 2]
```diff
@@ -74,30 +67,65 @@ def generate_metrics(self):
...
-        if self.associator is not None and self.association_set is None:
-            self.associate_tracks()
-
         metrics = {}
-        for generator in self.generators:
+
+        generators = self.generators if isinstance(self.generators, list) else [self.generators]
+
+        for generator in generators:
+            if isinstance(generator, SIAPMetrics):
```
Comment (sdhiscocks) [inline stonesoup/metricgenerator/manager.py:75]: Referencing to specific metric here doesn't seem appropriate. Why does this need to be done?

File: stonesoup/metricgenerator/manager.py [hunk 3]
```diff
@@ -74,30 +67,65 @@ def generate_metrics(self):
...
+
+        Returns
+        ----------
+        : dict of :class`SIAPMetrics` averages
+        """
+        siap_metrics = self.metrics[generator_name]
+        siap_averages = {siap_metrics.get(metric) for metric in siap_metrics
+                         if metric.startswith("SIAP") and not metric.endswith(" at times")}
+
+        return siap_averages
```
Comment (sdhiscocks) [inline stonesoup/metricgenerator/manager.py:131]: These methods also seem specific to the name or specific metric. A different approach is needed here.

**Pull request merged**